paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java

```diff
@@ -115,10 +114,20 @@ private TagAutoCreation(
         }
     }
 
+    public void refreshTags() {
```

Comment:
We should not add new method for `TagAutoCreation`.
We should just invoke `run`.

We can create a method:
```
private void tryToTag(Snapshot snapshot, LocalDateTime time) {
        if (nextTag == null
                || isAfterOrEqual(time.minus(delay), periodHandler.nextTagTime(nextTag))) {
            LocalDateTime thisTag = periodHandler.normalizeToTagTime(time);
            String tagName = periodHandler.timeToTag(thisTag);
            tagManager.createTag(snapshot, tagName);
            nextTag = periodHandler.nextTagTime(thisTag);

            if (numRetainedMax != null) {
                SortedMap<Snapshot, String> tags = tagManager.tags();
                if (tags.size() > numRetainedMax) {
                    int toDelete = tags.size() - numRetainedMax;
                    int i = 0;
                    for (String tag : tags.values()) {
                        tagManager.deleteTag(tag, tagDeletion, snapshotManager);
                        i++;
                        if (i == toDelete) {
                            break;
                        }
                    }
                }
            }
        }
    }
```

And in `run`:
```
    public void run() {
        while (true) {
            if (snapshotManager.snapshotExists(nextSnapshot)) {
                tryToTag(snapshotManager.snapshot(nextSnapshot));
                nextSnapshot++;
            } else {
                // avoid snapshot has been expired
                Long earliest = snapshotManager.earliestSnapshotId();
                if (earliest != null && earliest > nextSnapshot) {
                    nextSnapshot = earliest;
                } else {
                    break;
                }
            }
        }

        if (timeExtractor instanceof ProcessTimeExtractor) {
            tryToTag(snapshotManager.snapshot(nextSnapshot - 1), LocalDateTime.now());
        }
    }
```

Something like above codes.

Reply:
done.